packages/esm-forms-dashboard-app/src/hooks/use-forms.ts

```diff
@@ -0,0 +1,141 @@
+import dayjs from 'dayjs';
+import useSWR from 'swr';
+import { getDynamicOfflineDataEntries, openmrsFetch, useConfig, useVisit } from '@openmrs/esm-framework';
+import { ListResponse, Form, EncounterWithFormRef, CompletedFormInfo, FormsSection } from '../types';
+import { customEncounterRepresentation, formEncounterUrl, formEncounterUrlPoc } from '../constants';
+import { ConfigObject, FormsSectionConfig } from '../config-schema';
+
+export function useForms(cachedOfflineFormsOnly = false, patientUuid: string = '') {
+  const { showConfigurableForms, customFormsUrl, showHtmlFormEntryForms } = useConfig() as ConfigObject;
+  const url = showConfigurableForms
+    ? customFormsUrl.concat(`?patientUuid=${patientUuid}`)
+    : showHtmlFormEntryForms
+    ? formEncounterUrl
+    : formEncounterUrlPoc;
+
+  return useSWR([url, cachedOfflineFormsOnly], async () => {
+    const res = await openmrsFetch<ListResponse<Form>>(url);
+    // show published forms and hide component forms
+    const forms = showConfigurableForms
+      ? res?.data.results
+      : res.data?.results?.filter((form) => form.published && !/component/i.test(form.name)) ?? [];
+
+    if (!cachedOfflineFormsOnly) {
+      return forms;
+    }
+
+    const dynamicFormData = await getDynamicOfflineDataEntries('form');
+    return forms.filter((form) => dynamicFormData.some((entry) => entry.identifier === form.uuid));
+  });
+}
+
+export function useEncounters(patientUuid: string) {
+  const { useCurrentVisitDates } = useConfig() as ConfigObject;
+  const visits = useVisit(patientUuid);
+  let startDate = dayjs(new Date()).startOf('year').subtract(2, 'years').toDate().toISOString();
```

Comment:
Why two years? Isn't the right thing here to have `startDate` be `null` and just not add the parameter if has a value? Maybe instead of just doing string concatenation here, it would be clearer to use something like [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) to construct the query? That would allow you to programmatically construct the object and then just have:

```ts
return useSWR(`${encounterUrl}?${urlParams.toString()}` /* ... */)
```

Reply:
I have done it to keep a similar beavior of `esm-patient-forms-app`
```
startDate: Date = dayjs(new Date()).startOf('day').subtract(500, 'day').toDate(),
```

I just use 2 years instead of 500 days, it make more logic for me. I can try to remove it or try to use the 500 days. How should I proceed?